docs/DeveloperGuide.md

```diff
@@ -1,38 +1,321 @@
+    * [Data Component](#data-component)
+  * [**Feature**](#feature)
+    * [Add Dish](#add-dish)
+    * [Adding a Dish](#adding-a-dish)
+    * [List Menu](#list-menu)
+    * [Add Order](#add-order)
+    * [Next Day](#next-day)
+    * [Previous Day](#previous-day)
+    * [List Ingredients](#list-ingredients)
+    * [Delete Dish](#delete-dish)
+    * [Edit Price](#edit-price)
+    * [Help](#help)
+  * [**Product scope**](#product-scope)
+    * [Target user profile](#target-user-profile)
+    * [Value proposition](#value-proposition)
+    * [User stories](#user-stories)
+<!-- TOC -->
 
-## Acknowledgements
+--------------------------------------------------------------------------------------------------------------------
+## **Acknowledgements**
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+[addressbook-level2](https://github.com/se-edu/addressbook-level2) <br>
+[addressbook-level3](https://github.com/se-edu/addressbook-level3)
 
-## Design & implementation
+--------------------------------------------------------------------------------------------------------------------
+## **Setting up, getting started**
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Refer to the guide [_UserGuide_](UserGuide.md).
 
+--------------------------------------------------------------------------------------------------------------------    
+## **Design**
 
-## Product scope
-### Target user profile
+### Architecture
+![Architecture Diagram](images/ArchitectureDiagram.png)
 
-{Describe the target user profile}
+*Figure 1: Architecture Diagram*
 
-### Value proposition
+The ***Architecture Diagram*** given above explains the high-level design of the App.
+Listed below is a brief summary outlining the primary components and their interrelationships.
+
+In summary, the user interacts with the Ui components, initiating a sequence that involves:
+- `Parser` component for command interpretation
+- `Command` component for execution 
+- `Data` component for managing application data
+
+The `Storage` component mainly handles interaction with external text files and main coordinates the interactions between the various Components.
+
+The bulk of the app’s work is done by the following components:
+- `Ui` : The UI of the App.
+- `Storage` : Reads data from, and writes data to, the text files.
+- `Data` : Consists of all the classes that are involved in execution of commands.
+- `Parser` : Makes sense of user input to return the appropriate command
+- `Command` : Executes the command requested by the user.
+
+**How the architecture components interact with each other:**
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+
+![Architecture Encode Data](images/sequence/Architecture_Encode_Data.png)
```

Comment:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/101786149/7d1cadb1-13c0-414f-ad39-c989a8c7a595)
we should use the solid line.

Reply:
Hmmm I am not sure which part you are referring to, could you specific which statement is it?